service/dap/server.go

```diff
@@ -1254,6 +1258,81 @@ func (s *Server) onSetFunctionBreakpointsRequest(request *dap.SetFunctionBreakpo
 	s.send(response)
 }
 
+const instructionBpPrefix = "instructionBreakpoint"
+
+func (s *Server) onSetInstructionBreakpointsRequest(request *dap.SetInstructionBreakpointsRequest) {
+	// Get all existing function breakpoints.
+	existingBps := s.getMatchingBreakpoints(instructionBpPrefix)
+	bpAdded := make(map[string]struct{}, len(existingBps))
+	for _, bp := range existingBps {
+		existingBps[bp.Name] = bp
+	}
+
+	breakpoints := make([]dap.Breakpoint, len(request.Arguments.Breakpoints))
+	// Amend any existing breakpoints.
+	for i, want := range request.Arguments.Breakpoints {
+		reqString := fmt.Sprintf("%s PC=%s", instructionBpPrefix, want.InstructionReference)
+		var err error
+		got, ok := existingBps[reqString]
+		if !ok {
+			// Skip if the breakpoint does not already exist.
+			// These will be created after deleting existing
+			// breakpoints to avoid conflicts.
+			continue
+		}
+		if _, ok := bpAdded[reqString]; ok {
```

Comment:
Isn't this always false?

Reply:
This could be true if there are multiple instruction breakpoint requests with the same arguments.